osmoutils/cache_ctx.go

```diff
@@ -78,3 +77,16 @@ func PrintPanicRecoveryError(ctx sdk.Context, recoveryError interface{}) {
 	}
 	ctx.Logger().Error("stack trace: " + errStackTrace)
 }
+
+// NoEventCacheContext returns a cache context
+// The write method ignores the event manager
+func NoEventCacheContext(c sdk.Context) (cc sdk.Context, writeCache func()) {
```

Comment:
same as SDK's cacheCtx, only it's not emitting the events on `write()` call

Reply:
Yea I'm not sure you need this function at all.

what `EmitEvents` does is simply appending events.

```go 
func (em *EventManager) EmitEvents(events Events) {
	em.events = em.events.AppendEvents(events)
}
```

which I think you can get later with the `Events` and than modify. 

```go
func (em *EventManager) Events() Events { return em.events }
```

Reply:
I thought the same, but it's buggy
`cacheCtx, write := ctx.CacheContext()`

the `write()` still acts over the original ctx
I didn't find easier way to achieve it